lib/SILOptimizer/Utils/Devirtualize.cpp

```diff
@@ -516,7 +516,9 @@ bool swift::canDevirtualizeClassMethod(FullApplySite AI,
 
   // We need to disable the  “effectively final” opt if a function is inlinable
   if (isEffectivelyFinalMethod &&
-      F->getResilienceExpansion() == ResilienceExpansion::Minimal) {
+      ((F->getResilienceExpansion() == ResilienceExpansion::Minimal) ||
```

Comment:
Is the first check even necessary? It seems like this is always about the call site. cc @slavapestov

Reply:
I agree with @jrose-apple. It's always OK to inline an inlinable function into a non-inlinable function.

Reply:
Removed the first check: https://github.com/apple/swift/pull/17337